### PR TITLE
User role based access for APIs

### PR DESCRIPTION
**Pull request by pliablepixels**
@kylejohnson please review interim progress (Please label Not Ready to Merge) #1155 

So far I've implemented user roles for the Monitors API. If this approach is ok, I'll continue down this path.

### CODE REVIEW

File: web/api/app/Controller/AppController.php [hunk 1]
```diff
@@ -63,9 +63,47 @@ public function beforeFilter() {
...
+			{
+				throw new UnauthorizedException(__('User is not enabled'));
+				return;
+			}
+
+			if ($zmOptAuth=='1')
+			{
+				$options = array ('conditions' => array ('User.Username' => $loggedinUser));
+				$userMonitors = $this->User->find('first', $options);
+				$this->Session->Write('allowedMonitors',$userMonitors['User']['MonitorIds']);
```
Comment (kylejohnson) [inline web/api/app/Controller/AppController.php:88]: I haven't read up on this in a while, but what do you think of `Session->Write` vs. `Configure::write`?  I am using the latter in other places.

**Comment by kylejohnson** [comment]
Overall this looks pretty good.

Maybe I'm not reading the AppController class right - but how does a user actually log in to the API?

File: web/api/app/Controller/AppController.php (hunk 1 above)
Comment (pliablepixels) [inline web/api/app/Controller/AppController.php:88]: @kylejohnson , based on my reading:

`Configure::write` is for application wide variables while `Session->Write` is for settings that get set only for that specific session. The latter is what we need for these variables as they will change depending on who has opened the session

**Comment by pliablepixels** [comment]
@kylejohnson - the user does not log into the API. The user logs into ZM. This sets a set of variables - I am accessing those variables in the context of the same session to make sure the user has logged in.

**Comment by pliablepixels** [comment]
I'm closing this as PR https://github.com/ZoneMinder/ZoneMinder/pull/1196 supersedes this and adds OPT_API to UI too

**Pull request closed without merging**